packages/components/src/components/post-tooltip/post-tooltip.tsx

```diff
@@ -0,0 +1,178 @@
+import { Component, h, Element, Prop, Host, Method } from '@stencil/core';
+import { version } from '../../../package.json';
+import {
+  computePosition,
+  Placement,
+  flip,
+  shift,
+  autoUpdate,
+  offset,
+  inline,
+  arrow,
+} from '@floating-ui/dom';
+
+// Polyfill for popovers, can be removed when https://caniuse.com/?search=popover is green
+import '@oddbird/popover-polyfill';
+
+// Patch for long press on touch devices
+import 'long-press-event';
+
+const sidemap = {
+  top: 'bottom',
+  right: 'left',
+  bottom: 'top',
+  left: 'right',
+};
+
+interface IPopoverElement {
+  showPopover: () => void;
+  hidePopover: () => void;
+  togglePopover: (force?: boolean) => boolean;
+}
+
+@Component({
+  tag: 'post-tooltip',
+  styleUrl: 'post-tooltip.scss',
+  shadow: true,
+})
+export class PostTooltip {
+  @Element() host: HTMLPostTooltipElement;
+
+  /**
+   * Define the placement of the tooltip according to the floating-ui options available at https://floating-ui.com/docs/computePosition#placement.
+   * Tooltips are automatically flipped to the opposite side if there is not enough available space and are shifted
+   * towards the viewport if they would overlap edge boundaries.
+   */
+  @Prop() placement?: Placement = 'top';
+
+  /**
+   * Internally used to track changes to the class attribute on the host element
+   */
+  @Prop() class: string;
+
+  private tooltipRef: HTMLElement & IPopoverElement;
+  private arrowRef: HTMLElement;
+  private trigger: HTMLElement;
+  private clearAutoupdate: () => void;
+
+  componentWillLoad() {
+    this.trigger = document.querySelector(`[data-tooltip-target="${this.host.id}"]`);
+
+    // Patch popovertargetaction="interest" until it's implemented
+    // https://github.com/openui/open-ui/issues/767#issuecomment-1654177227
+    this.trigger.addEventListener('mouseenter', this.showTooltip.bind(this));
+    this.trigger.addEventListener('mouseleave', this.hideTooltip.bind(this));
+    this.trigger.addEventListener('focus', this.showTooltip.bind(this));
+    this.trigger.addEventListener('blur', this.hideTooltip.bind(this));
+    this.trigger.addEventListener('long-press', this.showTooltip.bind(this));
+
+    // Patch missing aria-describedby attribute on the trigger
+    const describedBy = this.trigger.getAttribute('aria-describedby') || '';
+    if (!describedBy?.includes(this.host.id)) {
+      this.trigger.setAttribute(
+        'aria-describedby',
+        [...describedBy.split(' '), this.host.id].join(' '),
+      );
+    }
+  }
+
+  componentDidRender() {
+    // Can't figure out how to extend HTMLAttributes<HTMLParagraphElement> to support the popover attribute
+    this.tooltipRef.setAttribute('popover', '');
+  }
```

Comment:
This is a really tricky TS situation where builds fail if a `<p popover>` is present. Temporary workaround, posted a question here https://discord.com/channels/520266681499779082/1145691780625473628/1145691780625473628 and will update when answered.

Reply:
Is this still an issue?